libs/sdk-ui-dashboard/src/drill/DrillSelect/WithDrillSelect.tsx

```diff
@@ -0,0 +1,168 @@
+        onDrillDown?.({
+            drillDefinition: s.payload.drillDefinition,
+            drillEvent: s.payload.drillEvent,
+            insight: s.payload.insight,
+        });
+    }, handleError);
+
+    const drillToInsight = useDrillToInsight((s) => {
+        onDrillToInsight?.({
+            drillDefinition: s.payload.drillDefinition,
+            drillEvent: s.payload.drillEvent,
+            insight: s.payload.insight,
+        });
+    }, handleError);
+
+    const drillToDashboard = useDrillToDashboard((s) => {
+        onDrillToDashboard?.({
+            drillDefinition: s.payload.drillDefinition,
+            drillEvent: s.payload.drillEvent,
+            filters: s.payload.filters,
+        });
+    }, handleError);
+
+    const drillToAttributeUrl = useDrillToAttributeUrl((s) => {
+        onDrillToAttributeUrl?.({
+            drillDefinition: s.payload.drillDefinition,
+            drillEvent: s.payload.drillEvent,
+            url: s.payload.url,
+        });
+    }, handleError);
+
+    const drillToCustomUrl = useDrillToCustomUrl((s) => {
+        onDrillToCustomUrl?.({
+            drillDefinition: s.payload.drillDefinition,
+            drillEvent: s.payload.drillEvent,
+            url: s.payload.url,
+        });
+    }, handleError);
+
+    const onSelect = useCallback(
+        (drillDefinition: DashboardDrillDefinition, drillEvent?: IDashboardDrillEvent) => {
+            const effectiveDrillEvent = drillEvent ?? dropdownProps?.drillEvent;
+            if (effectiveDrillEvent) {
+                if (isDrillDownDefinition(drillDefinition)) {
+                    drillDown.run(insight, drillDefinition, effectiveDrillEvent);
+                } else if (isDrillToInsight(drillDefinition)) {
+                    drillToInsight.run(drillDefinition, effectiveDrillEvent);
+                } else if (isDrillToDashboard(drillDefinition)) {
+                    drillToDashboard.run(drillDefinition, effectiveDrillEvent);
+                } else if (isDrillToAttributeUrl(drillDefinition)) {
+                    drillToAttributeUrl.run(drillDefinition, effectiveDrillEvent);
+                } else if (isDrillToCustomUrl(drillDefinition)) {
+                    drillToCustomUrl.run(drillDefinition, effectiveDrillEvent);
+                }
+                setDropdownProps(null);
+                setIsOpen(false);
+            }
+        },
+        [dropdownProps],
+    );
+
+    const drill = useDrill((s) => {
+        if (s.payload.drillEvent.drillDefinitions!.length === 1) {
+            onSelect(s.payload.drillEvent.drillDefinitions![0], s.payload.drillEvent);
+        } else {
+            setDropdownProps({
+                drillDefinitions: s.payload.drillEvent.drillDefinitions!,
+                drillEvent: s.payload.drillEvent,
+                drillContext: s.payload.drillContext,
+            });
+            setIsOpen(true);
+        }
+    }, handleError);
+
+    useEffect(() => {
+        setDrillPickerId(uuid());
```

Comment:
This is the only place this is set, do we really need a state for this? This thing could probably be replaced with

```ts
const drillPickerId = useRef(uuid());
```